ocf_datapipes/load/nwp/nwp.py

```diff
@@ -53,9 +93,46 @@ def __init__(
         else:
             raise ValueError(f"Unknown provider: {provider}")
 
-    def __iter__(self) -> Union[xr.DataArray, xr.Dataset]:
+    def __iter__(self) -> Union[xr.DataArray, xr.Dataset]:  # type: ignore
         """Opens the NWP data"""
         logger.debug("Opening NWP data: %s", self.zarr_path)
         nwp = self.open_nwp(self.zarr_path)
+        if self.check_for_zeros:
+            self.check_if_zeros(nwp)
+        if self.check_physical_limits:
+            self.check_if_physical_limits(nwp)
         while True:
             yield nwp
+
+    def check_if_zeros(self, nwp: Union[xr.DataArray, xr.Dataset]):
+        """Checks if the NWP data contains zeros"""
+        if isinstance(nwp, xr.DataArray):
+            if (nwp.values == 0).any():
```

Comment:
This looks like it will not be performed lazily (as in, this will load the whole dataArray into memory to check the values), which we really want to avoid in this place because at this point the arrays we operate on are often massive

Reply:
I've made some changes to accommodate lazy loading. Have leveraged Dask arrays, as its often used with xarray for the same

Reply:
That's a good way to go about it! There is a danger that some of our data might not fit anyway, but since it can be turned on and off that's fine.

I was wondering if it's worth exploring implementing this check downstream, somewhere after spacial and temporal crop and before normalisation, so that it operates on samples instead? And then maybe skip ones with too many zeroes/nans/out of physical bounds values and give a userWarning/log info of how many were skipped as a proxy for understanding how much of the data is corrupted. Thoughts @Sukh-P @peterdudfield?

Reply:
I like that idea, less chance of a chance of running into memory issues by loading a chunk, I guess the only draw back would be doing processing on some data you are going to chuck anyway but in this case that processing gets it down to a more manageable size

Reply:
@AUdaltsova I'm trying to understand if this is acceptable w.r.t the scope of this PR? 🤔

Reply:
@glitch401 might be! @peterdudfield happy to merge this then? :)